src/driver/xen.rs

```diff
@@ -49,6 +57,42 @@ impl Xen {
 
 impl api::Introspectable for Xen {
 
+    fn read_physical(&self, paddr: u64, count: u32) -> Result<Vec<u8>,&str> {
+        let mut cur_paddr: u64;
+        let mut offset: u64 = 0;
+        let mut read_len;
+        let mut count_mut: u64 = count.into();
+        let mut buf_offset = 0;
+        while count_mut > 0 {
+            // compute new paddr
+            cur_paddr = paddr + offset;
+            // get the current gfn
+            let gfn = cur_paddr >> xenctrl::PAGE_SHIFT;
+            offset = ((xenctrl::PAGE_SIZE - 1) as u64) & cur_paddr;
+            // map gfn
+            let map = self.xen_fgn.map(self.domid, PROT_READ, gfn).unwrap() as *mut c_char;
+            // determine how much we can read
+            if (offset + count as u64) > xenctrl::PAGE_SIZE as u64 {
+                read_len = (xenctrl::PAGE_SIZE as u64) - offset;
+            } else {
+                read_len = count_mut.into();
+            }
+
+            // convert to u8 array
+            let page = unsafe {
```

Comment:
@tathanhdinh idealy, what we should do here is move `std::slice::from_raw_parts_mut` to the `map` call in `xenforeignmemory`, so that `map` returns an `&mut [u8]`.

However, how do we retrieve the pointer afterwards to unmap the page ?